.github/actions/setup-nuget-sources/action.yml

```diff
@@ -0,0 +1,22 @@
+name: 'Setup NuGet Sources'
+description: 'Configure NuGet sources for the project'
+
+inputs:
+  github-token:
+    description: 'GitHub token for authentication'
+    required: true
+  github-username:
+    description: 'GitHub username'
+    required: true
+  
+runs:
+  using: 'composite'
+  steps:
+    - name: Add NuGet sources
+      shell: bash
+      env:
+        DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+        DOTNET_CLI_TELEMETRY_OPTOUT: 1
+      run: |
+        dotnet nuget add source --name github-codeo --username '${{ inputs.github-username }}' --password '${{ inputs.github-token }}' --store-password-in-clear-text https://nuget.pkg.github.com/codeo-za/index.json
```

Comment:
Same. If you used any tokens to run this stuff, you'll need to re-generate them. Its not safe to have called workflow actions with secrets.